stream-chat-android-state/src/main/java/io/getstream/chat/android/offline/plugin/logic/channel/internal/ChannelStateLogicImpl.kt

```diff
@@ -346,12 +346,19 @@ internal class ChannelStateLogicImpl(
     /**
      * Deletes a member. Doesn't delete in the database.
      *
-     * @param userId Id of the user.
+     * @param member The member to be removed.
      */
-    override fun deleteMember(userId: String) {
-        mutableState.rawMembers = mutableState.rawMembers - userId
+    override fun deleteMember(member: Member) {
+        val user = member.user
+
+        mutableState.rawMembers = mutableState.rawMembers - user.id
 
         mutableState.setMembersCount(mutableState.membersCount.value - 1)
+
+        if (user.online) {
+            deleteWatcher(user)
+            setWatcherCount(mutableState.watcherCount.value - 1)
+        }
```

Comment:
Hey @bychkovdmitry,

I wonder if we could use a `member` object available in `mutableState.rawMembers`:
```kotlin
    override fun deleteMember(userId: String) {
        val member = mutableState.rawMembers[userId]
        val user = member.user
        ...
```

Reply:
I'm not 100% sure that the current user will be in the first page of members

Reply:
There is no guarantee for that